src/include/daos_task.h

```diff
@@ -173,12 +173,14 @@ typedef struct {
 
 /** pool connect args */
 typedef struct {
-	/** UUID of the pool. */
-	uuid_t			uuid;
+	/** Label of the pool. May be NULL if connecting by UUID */
+	const char		*label;
+	/** UUID of the pool. May be the NULL UUID if connecting  by label */
+	uuid_t			 uuid;
```

Comment:
In case it is necessary, I have made a change in my local repo to have a daos_pool_connect_bylabel_t type, and refactored the dc_pool_connect() and (new) dc_pool_connect_bylabel() so that they can still use a lot of common code without copy/paste (allocate / init struct dc_pool and perform the rest of the pool connect procedure).